Megaparsec/Ok.lean

```diff
@@ -6,13 +6,13 @@ open Megaparsec.ParserState
 
 namespace Megaparsec.Ok
 
-def Ok (m : Type f → Type v) (β ℘ E γ : Type u) (ξ : Type f) :=
+def Ok (m : Type f → Type v) (β ℘ E γ : Type u) (ξ : Type f) [Ord β] [Ord E] :=
   (γ → State β ℘ E → Hints β → m ξ)
 
 /-
 `accHints hs c` adds hints to “OK” continuation that will add given
 hints @hs@ to third argument of original continuation c
 -/
-def accHints (hs : Hints β)
+def accHints [Ord β] [Ord E] (hs : Hints β)
```

Comment:
In some places you have constraints in the end of argument list, in some -- in the beginning. Can you explain this decision or is it arbitrary?